doc/modules/grid_search.rst

```diff
@@ -231,6 +231,196 @@ some parameter settings could be fully evaluated. Setting ``error_score=0``
 warning and setting the score for that fold to 0 (or `NaN`), but completing
 the search.
 
+
+.. _successive_halving_user_guide:
+
+Searching optimal parameters with successive halving
+====================================================
+
+Scikit-learn also provides the :class:`GridHalvingSearchCV` and
+:class:`RandomHalvingSearchCV` estimators that can be used to
+search a parameter space using successive halving [1]_ [2]_. Successive
+halving is an iterative selection process where all candidates are evaluated
+with a small amount of resources at the first iteration. Only a subset of
+these candidates are selected for the next iteration, which will be
+allocated more resources. What defines a resource is typically the number of
+samples to train on, or the number of trees for a gradient boosting /
+decision forest estimator.
+
+As illustrated in the figure below, only a small subset of candidates 'survive'
+until the last iteration. These are the candidates that have consistently been
+part of the best candidates across all iterations.
+
+.. figure:: ../auto_examples/svm/images/sphx_glr_plot_successive_halving_iterations_001.png
+   :target: ../auto_examples/model_selection/plot_successive_halving_iterations.html
+   :align: center
+
+The amount of resources ``r_i`` allocated for each candidate at iteration
+``i`` is controlled by the parameters ``ratio`` and ``r_min`` as follows::
+
+    r_i = ratio**i * r_min
```

Comment:
I think it would be helpful to expand and explain that:
"So in the first iteration, we use `r_min` resources `n_candidates` times. In the second iteration, we use `r_min * ratio` resources `n_candidates / ratio` times. The third again multiplies the resources per candidate and divides the number of candidates. This process stops when the maximum resources[/budget] per candidate is reached."

Reply:
Thanks, I used your comment and also added a table to examplify